src/bun.js/node/node_cluster_binding.zig

```diff
@@ -0,0 +1,280 @@
+const std = @import("std");
+const bun = @import("root").bun;
+const Environment = bun.Environment;
+const JSC = bun.JSC;
+const string = bun.string;
+const Output = bun.Output;
+const ZigString = JSC.ZigString;
+const log = Output.scoped(.IPC, false);
+
+extern fn Bun__Process__queueNextTick1(*JSC.JSGlobalObject, JSC.JSValue, JSC.JSValue) void;
+extern fn Process__emitErrorEvent(global: *JSC.JSGlobalObject, value: JSC.JSValue) void;
+
+pub var child_singleton: InternalMsgHolder = .{};
+
+pub fn sendHelperChild(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(JSC.conv) JSC.JSValue {
+    log("sendHelperChild", .{});
+
+    const arguments = callframe.arguments(3).ptr;
+    const message = arguments[0];
+    const handle = arguments[1];
+    const callback = arguments[2];
+
+    const vm = globalThis.bunVM();
+
+    if (vm.ipc == null) {
+        return .false;
+    }
+    if (message.isUndefined()) {
+        return globalThis.throwValueRet(globalThis.ERR_MISSING_ARGS_static(ZigString.static("message"), null, null));
+    }
+    if (!handle.isNull()) {
+        globalThis.throw("passing 'handle' not implemented yet", .{});
+        return .zero;
+    }
+    if (!message.isObject()) {
+        return globalThis.throwValueRet(globalThis.ERR_INVALID_ARG_TYPE_static(
+            ZigString.static("message"),
+            ZigString.static("object"),
+            message,
+        ));
+    }
+    if (callback.isFunction()) {
+        child_singleton.callbacks.put(bun.default_allocator, child_singleton.seq, JSC.Strong.create(callback, globalThis)) catch bun.outOfMemory();
+    }
+
+    // sequence number for InternalMsgHolder
+    message.put(globalThis, ZigString.static("seq"), JSC.JSValue.jsNumber(child_singleton.seq));
```

Comment:
i still don't like that we do this

these messages should be 2 int32's and nothing else, there's no variable-length or type data in these internal messages afaict